packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -245,8 +264,14 @@ export class WindowsOS extends OperatingSystem {
  * OS features specialized for Linux
  */
 export class LinuxOS extends OperatingSystem {
-  public createUserData(scripts: string[]): string {
-    return '#!/bin/bash\n' + scripts.join('\n');
+  public createUserData(scripts: string[], options: UserDataOptions): string {
+    const bashFlags = [];
+    if (options.verbose !== false) { bashFlags.push('x'); }
+    if (options.strict) { bashFlags.push('eu'); }
+
+    const bashCommand = '#!/bin/bash' + (bashFlags.length > 0) ? ` -${bashFlags.join('')}` : '';
```

Comment:
Are you sure it's possible to pass in flags like this instead of `set -euo pipefail...`

Reply:
Yes:

```
huijbers ~/W/P/ecs-cdk(ecs-demo)> bash -euo pipefail
bash-3.2$ echo $HELLO
bash: HELLO: unbound variable
huijbers ~/W/P/ecs-cdk(ecs-demo)> 
```